src/Compilers/VisualBasic/Test/Semantic/Semantics/NameOfTests.vb

```diff
@@ -3521,5 +3521,21 @@ N1N2
 ]]>).VerifyDiagnostics()
         End Sub
 
+        <Fact, WorkItem(564, "https://github.com/dotnet/roslyn/issues/564")>
+        Public Sub NameOfTypeParameterInDefaultValue()
+            Dim compilationDef =
+<compilation>
+    <file name="a.vb">
+Module Program
+    Sub M(Of TP)(Optional name As String = NameOf(TP))
```

Comment:
Consider adding Sub Main that calls this function without arguments, printing the value within this function and verifying expected output in CompileAndVerify call.